src/HieDb/Create.hs

```diff
@@ -149,14 +153,20 @@ addArr (getConn -> conn) arr = do
     addName :: Name -> IO (Maybe Int64)
     addName n = case nameModule_maybe n of
       Nothing -> pure Nothing
-      Just m -> liftIO $ do
+      Just m -> do
         let occ = nameOccName n
             mod = moduleName m
             uid = moduleUnitId m
         execute conn "INSERT INTO typenames(name,mod,unit) VALUES (?,?,?)" (occ,mod,uid)
-        Just . fromOnly . head <$> query conn "SELECT id FROM typenames WHERE name = ? AND mod = ? AND unit = ?" (occ,mod,uid)
-
-addTypeRefs :: HieDb -> FilePath -> HieFile -> A.Array TypeIndex (Maybe Int64) -> IO ()
+        Just <$> lastInsertRowId conn
```

Comment:
[lastInsertRowId](https://hackage.haskell.org/package/sqlite-simple-0.4.18.0/docs/Database-SQLite-Simple.html#v:lastInsertRowId) is more efficient way to achieve the same.